internal/spdxlicense/generate_license_list.go

```diff
@@ -0,0 +1,92 @@
+// +build ignore
```

Comment:
what's the purpose of this code?

Reply:
SPDX spec dictates which authoritative license names can be reported out in an SPDX SBOM. They maintain the authoritative list at https://spdx.org/licenses/licenses.json and the version of this list needs to be additionally reported in the SBOM. This `generate_license_list.go` responds to the `go generate ./...` command in this repo based on the `//go:generate go run generate_license_list.go` comment in `license.go`. 

Specifically the `// +build ignore` comment in `generate_license_list.go` is meant to keep this go source file out of the syft build. That is, when running `go build ./...` this "build tag" is used to tell the compiler to skip this file altogether when making the syft binary. The `generate_license_list.go` file is still kept specially near the file it is generating for simplicity (`license_list.go`).

So in order to build syft with the latest license list (pulled from https://spdx.org/licenses/licenses.json) one needs to run:
```
go generate ./...
go build ./...
```

But if you want to keep the existing license list and just build syft then it you do the same as you would today:
```
go build ./...
```

note: I've got an integration test that checks to see if there is a later version of the license list than what we have now --if there is then the test fails and notes in the output as to why.